### PR TITLE
Feature/open rpc

### PR DESCRIPTION
**Pull request by wow-sven**
Part of #213 

discussion: https://discord.com/channels/1078938449974935592/1118135406530150451

### CODE REVIEW

File: crates/rooch-server/src/jsonrpc_types/move_types.rs [hunk 1]
```diff
@@ -325,8 +329,221 @@ impl FromStr for StrView<ModuleId> {
...
+    }
+}
+
+impl From<AccessPath> for AccessPathView {
+    fn from(value: AccessPath) -> Self {
+        AccessPathView(value.0.into())
+    }
+}
+
+// TODO: Specifies whether to merge Moveos h256 and rooch h256
```
Comment (jolestar) [inline crates/rooch-server/src/jsonrpc_types/move_types.rs:451]: I think can merge the two into H256View, or directly use StrView<H256>

File: crates/rooch-server/src/jsonrpc_types/move_types.rs [hunk 2]
```diff
@@ -325,8 +329,221 @@ impl FromStr for StrView<ModuleId> {
...
+                account: account.into(),
+                module_names: module_names.into_iter().map(IdentifierView::from).collect(),
+            },
+            Path::Table { table_handle, keys } => Self::Table { table_handle, keys },
+        }
+    }
+}
+
+#[derive(Debug, Clone, Deserialize, Serialize, JsonSchema)]
+pub struct AccessPathView(pub PathView);
```
Comment (jolestar) [inline crates/rooch-server/src/jsonrpc_types/move_types.rs:437]: Maybe AccessPathView can simply be defined via StrView<AccessPath>

**Review by jolestar — APPROVED** [review]
LGTM @baichuan3 take a look.

File: crates/rooch-server/src/api/rooch_api.rs [hunk 3]
```diff
@@ -2,68 +2,73 @@
...
+        &self,
+        filter: EventFilterView,
+    ) -> RpcResult<Vec<Option<AnnotatedEventView>>>;
 
-    #[method(name = "rooch_getTransactionByHash")]
-    async fn get_transaction_by_hash(&self, hash: H256) -> RpcResult<Option<TransactionView>>;
+    #[method(name = "getTransactionByHash")]
+    async fn get_transaction_by_hash(
+        &self,
+        hash: RoochH256View,
```
Comment (baichuan3) [inline crates/rooch-server/src/api/rooch_api.rs:68]: Do all non-primitive types must  be View types or are they optional?

**Review by baichuan3 — APPROVED** [review]
LGTM

File: crates/rooch-server/src/api/rooch_api.rs (hunk 3 above)
Comment (jolestar) [reply crates/rooch-server/src/api/rooch_api.rs:68]: View just for better json display. We can discuss on the next dev meeting.

**Comment by jolestar** [comment]
I merge first. @wow-sven you can address the comments in the next PR. @wow-sven

**Pull request merged**